api/rosetta/data_block_integration_test.go

```diff
@@ -0,0 +1,563 @@
+			wantRosettaError:            configuration.ErrorUnknownBlock,
+			wantRosettaErrorDescription: fmt.Sprintf("block index is above last indexed block (last: %d)", lastKnownBlockHeight),
+			wantRosettaErrorDetails:     map[string]interface{}{"index": uint64(426), "hash": ""},
+		},
+		{
+			name: "mismatched block height and hash",
+			request: rosetta.BlockRequest{
+				NetworkID: defaultNetworkID(),
+				BlockID: identifier.Block{
+					Index: 43,
+					Hash:  knownBlockID(44),
+				},
+			},
+
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidBlock,
+			wantRosettaErrorDescription: fmt.Sprintf("block hash does not match known hash for height (known: %s)", knownBlockID(43)),
+			wantRosettaErrorDetails:     map[string]interface{}{"index": uint64(43), "hash": knownBlockID(44)},
+		},
+		{
+			// effectively the same as the 'missing blockchain name' test case, since it's the first check we'll do
+			name:    "empty block request",
+			request: rosetta.BlockRequest{},
+
+			wantStatusCode:              http.StatusBadRequest,
+			wantRosettaError:            configuration.ErrorInvalidFormat,
+			wantRosettaErrorDescription: "blockchain identifier: blockchain field is empty",
+			wantRosettaErrorDetails:     nil,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			enc, err := json.Marshal(test.request)
+			require.NoError(t, err)
+
+			req := httptest.NewRequest(http.MethodPost, "/block", bytes.NewReader(enc))
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			rec := httptest.NewRecorder()
+			ctx := echo.New().NewContext(req, rec)
+
+			// execute the request
+			err = api.Block(ctx)
+			assert.Error(t, err)
+
+			echoErr, ok := err.(*echo.HTTPError)
+			require.True(t, ok)
+
+			// verify HTTP status code
+			assert.Equal(t, test.wantStatusCode, echoErr.Code)
+
+			gotErr, ok := echoErr.Message.(rosetta.Error)
+			require.True(t, ok)
+
+			assert.Equal(t, test.wantRosettaError, gotErr.ErrorDefinition)
+			assert.Equal(t, test.wantRosettaErrorDescription, gotErr.Description)
+			assert.Equal(t, test.wantRosettaErrorDetails, gotErr.Details)
+		})
+	}
+}
+
+func TestMalformedBlockRequest(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	tests := []struct {
+		name     string
+		payload  []byte
+		mimeType string
+	}{
+		{
+			name:     "empty request",
+			payload:  []byte(``),
+			mimeType: echo.MIMEApplicationJSON,
+		},
+		{
+			name:     "wrong field type",
+			payload:  []byte(`{ "network_identifier": { "blockchain": "flow", "network": 99} }`),
+			mimeType: echo.MIMEApplicationJSON,
+		},
+		{
+			name: "unclosed bracket",
+			payload: []byte(`{ "network_identifier": { "blockchain": "flow", "network": "flow-testnet" },
+							   "block_identifier": { "index": 13, "hash": "af528bb047d6cd1400a326bb127d689607a096f5ccd81d8903dfebbac26afb23" }`),
```

Comment:
Maybe we could define those with proper white space / indentation as constant strings before the tests, and then just use them?

```
payload: unclosedBracked,

payload: wrongFieldType,
```
And so on?